src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +14,189 @@ ${json}
 \`\`\``;
 };
 
+const StorageDomain = {
+  page: "confirm_transactions",
+};
+
+const StorageType = {
+  SendTransactionWithoutConfirmation: "send_transaction_without_confirmation",
+};
+
 export default function ConfirmTransactions(props) {
   const gkey = useState(uuid());
   const near = useNear(props.networkId);
+  const accountId = useAccountId(props.networkId);
+  const cache = useCache();
+
   const [loading, setLoading] = useState(false);
 
+  const [transactions] = useState(props.transactions);
+  const [dontAskForConfirmation, setDontAskForConfirmation] = useState(null);
+  const [dontAskAgainChecked, setDontAskAgainChecked] = useState(false);
+
+  const widgetSrc = props.widgetSrc;
+
+  const getWidgetContractPermission = async (widgetSrc, contractId) =>
+    await cache.asyncLocalStorageGet(StorageDomain, {
+      widgetSrc,
+      contractId,
+      type: StorageType.SendTransactionWithoutConfirmation,
+    });
+
+  useEffect(() => {
+    (async () => {
+
+      if (transactions.length == 1) {
+        const contractId = transactions[0].contractName;
+        const isSignedIntoContract = await near.isSignedIntoContract(contractId);
+
+        const widgetContractPermission = await getWidgetContractPermission(widgetSrc, contractId);
+
+        const dontAskForConfirmation = (isSignedIntoContract && widgetContractPermission && widgetContractPermission[transactions[0].methodName])
+          ? true : false;
+
+        setDontAskForConfirmation(dontAskForConfirmation);
+
+        if (dontAskForConfirmation) {
+          setLoading(true);
+          const result = await near.sendTransactions(transactions);
+          setLoading(false);
+          onHide(result);
+        }
+      }
+    })();
+  }, []);
+
   const onHide = props.onHide;
-  const transactions = props.transactions;
+
   const show = !!transactions;
 
-  return (
-    <Modal size="xl" centered scrollable show={show} onHide={onHide}>
-      <Modal.Header closeButton>
-        <Modal.Title>Confirm Transaction</Modal.Title>
+  const dontAskAgainCheckboxChange = async () => {
+    setDontAskAgainChecked(!dontAskAgainChecked);
+  };
+
+  if (dontAskForConfirmation) {
+    const transaction = transactions[0];
+    return (<Modal size="xs" centered show={show} onHide={onHide}>
+      <Modal.Header>
+        <Modal.Title>Sending transaction</Modal.Title>
```

Comment:
I feel it is quite distracting to show it as a modal. Ideally, I would use a [toast](https://getbootstrap.com/docs/5.3/components/toasts/).

Reply:
changed to using a toast